src/Apps/Order2/Routes/Checkout/Components/CriticalErrorModal.tsx

```diff
@@ -0,0 +1,59 @@
+import { Button, Flex, ModalDialog, Text } from "@artsy/palette"
+import { useReturnToArtwork } from "Apps/Order2/Routes/Checkout/Hooks/useReturnToArtwork"
+
+export const CriticalErrorModal: React.FC<{
+  error: string | null
+}> = ({ error }) => {
+  const returnToArtwork = useReturnToArtwork()
+
+  if (!error) {
+    return null
+  }
+
+  const handleClose = () => {
+    returnToArtwork()
+  }
+
+  const handleReload = () => {
+    window.location.reload()
+  }
+
+  // Determine if reload should be available based on error type
+  let canReload = false
+
+  const title = "Checkout Error"
+  let description: string
+
+  switch (error) {
+    case "loading_timeout":
+      description =
+        "There was an error loading your checkout. Please try refreshing the page."
```

Comment:
#minor. Wonder if we should standardize on the term: refresh vs. reload (as in the CTA).